src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -257,6 +257,11 @@ private boolean isFuelNeeded()
             if (WorldUtil.isBlockLoaded(world, pos))
             {
                 final TileEntity entity = world.getTileEntity(pos);
+                if(!(entity instanceof FurnaceTileEntity))
+                {
+                    getOwnBuilding().removeFromFurnaces(pos);
+                    continue;
```

Comment:
I wonder if this could cause a concurrent modification exception. Are we iterating over a copy?

Reply:
Just tested this scenario. I didn't get an exception.